templates/demo/app/components/storage.js

```diff
@@ -1,172 +1,241 @@
-                    <HelpBlock>generated hash: <span className="fileHash">{this.state.fileHash}</span></HelpBlock>
-                </FormGroup>
-            </Form>
-        
-            <h3>Get file or image from storage</h3>
-            <Form inline>
-                <FormGroup>
-                    <FormControl
-                        type="text"
-                        value={this.state.imageToDownload}
-                        onChange={e => this.handleChange(e, 'imageToDownload')} />
-                    <Button bsStyle="primary" onClick={(e) => this.loadFile(e)}>Download</Button>
-                    <HelpBlock>file available at: <span><a href={this.state.url} target="_blank">{this.state.url}</a></span></HelpBlock>
-                    <HelpBlock><img src={this.state.url} /></HelpBlock>
-                </FormGroup>
-            </Form>
-
-            <p>Javascript calls being made: </p>
-            <div className="logs">
-            <p>EmbarkJS.Storage.setProvider('ipfs',{'{'}server: 'localhost', port: '5001'{'}'})</p>
-            {
-                this.state.logs.map((item, i) => <p key={i}>{item}</p>)
-            }
-            </div>
-        </React.Fragment>;
-    }
+        });
+        this.addToLog("EmbarkJS.Storage.saveText('" + this.state.textToSave + "').then(function(hash) { })");
+      })
+      .catch((err) => {
+        if (err) {
+          this.setState({storageError: err.message});
+          console.log("Storage saveText Error => " + err.message);
+        }
+      });
+  }
+
+  loadHash(e) {
+    e.preventDefault();
+
+    EmbarkJS.Storage.get(this.state.loadText)
+      .then((content) => {
+        this.setState({storedText: content, storageError: ''});
+        this.addToLog("EmbarkJS.Storage.get('" + this.state.loadText + "').then(function(content) { })");
+      })
+      .catch((err) => {
+        if (err) {
+          this.setState({storageError: err.message});
+          console.log("Storage get Error => " + err.message);
+        }
+      });
+  }
+
+  uploadFile(e) {
+    e.preventDefault();
+
+    EmbarkJS.Storage.uploadFile(this.state.fileToUpload)
+      .then((hash) => {
+        this.setState({
+          fileHash: hash,
+          imageToDownload: hash,
+          storageError: ''
+        });
+        this.addToLog("EmbarkJS.Storage.uploadFile(this.state.fileToUpload).then(function(hash) { })");
+      })
+      .catch((err) => {
+        if (err) {
+          this.setState({storageError: err.message});
+          console.log("Storage uploadFile Error => " + err.message);
+        }
+      });
+  }
+
+  loadFile(e) {
+    let _url = EmbarkJS.Storage.getUrl(this.state.imageToDownload);
+    this.setState({url: _url});
+    this.addToLog("EmbarkJS.Storage.getUrl('" + this.state.imageToDownload + "')");
+  }
+
+  ipnsRegister(e) {
+    e.preventDefault();
+    this.setState({ ipnsRegistering: true });
+    EmbarkJS.Names.register(this.state.ipfsHash, (err, name) => {
+      if (err) {
```

Comment:
If you can, maybe add a visual indicator that there was an error (like putting the error message visually with red text). You can check the ENS component. I put an error component when register and others error.